pages/[wallet]/index.tsx

```diff
@@ -223,56 +223,94 @@ export default function Dashboard() {
     arbitrumTestnetQuery.data,
   ]);
 
+  const isFetched =
+    mainnetQuery.isFetched &&
+    rinkebyQuery.isFetched &&
+    goerliQuery.isFetched &&
+    polygonQuery.isFetched &&
+    mumbaiQuery.isFetched &&
+    fantomQuery.isFetched &&
+    fantomTestnetQuery.isFetched &&
+    avalancheQuery.isFetched &&
+    avalancheFujiTestnetQuery.isFetched &&
+    optimismQuery.isFetched &&
+    optimismTestnetQuery.isFetched &&
+    arbitrumQuery.isFetched &&
+    arbitrumTestnetQuery.isFetched;
+
+  useEffect(() => {
+    if (isFetched && combinedList.length === 0) {
+      router.replace("/contracts");
+    }
+  }, [isFetched, router, combinedList]);
+
```

Comment:
we're still not considering the loading state of v1 projects

Reply:
also big oof at this logic list.

Reply:
agree